### PR TITLE
[flutter_tools] switch to package test coverage, remove build_runner

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove tooling usage of build_runner for testing, and tooling specific coverage collector. Use the package:test coverage collector and package:coverage formatter.

This does not noticeably increase CI times, and allows us to further reduce the number of tooling dependencies

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -172,7 +172,7 @@ task:
         - dart --enable-asserts ./dev/bots/test.dart
 
     - name: tool_coverage-linux # linux-only
-      only_if: "$CIRRUS_BRANCH == 'master'"
+      #only_if: "$CIRRUS_BRANCH == 'master'"
```
Comment (jonahwilliams) [inline .cirrus.yml]: Temporarily testing coverage performance

File: dev/bots/test.dart [hunk 2]
```diff
@@ -706,25 +703,15 @@ Future<void> _runFlutterWebTest(String workingDirectory, List<String> tests) asy
...
-      '--',
-    ]);
-  } else {
-    args.add('test');
-  }
+  final List<String> args = <String>['run', 'test'];
   args.add(useFlutterTestFormatter ? '-rjson' : '-rcompact');
+  if (coverage != null)
+    args.add('--coverage=$coverage');
   int cpus;
```
Comment (zanderso) [inline dev/bots/test.dart]: Cleanup: Hoist the `cpus` calculation and switch to the new list literal syntax.

**Review by zanderso — COMMENTED** [review]
Why is the tool-specific coverage collector obsolete? Were your improvements upstreamed?

**Comment by jonahwilliams** [comment]
Yup! test now has a --coverage argument and it is fairly quick

**Comment by jonahwilliams** [comment]
This has a bad merge conflict, trying again here: https://github.com/flutter/flutter/pull/51966

**Pull request closed without merging**